imap_processing/utils.py

```diff
@@ -94,3 +99,64 @@ def convert_raw_to_eu(dataset: xr.Dataset, conversion_table_path, packet_name):
         )
 
     return dataset
+
+
+def create_dataset(packets: list[Packet], met_name="shcoarse") -> xr.Dataset:
+    """Create dataset for each metadata field.
+
+    # TODO: change IMAP-Hi to use this and others as needed.
+
+    Parameters
+    ----------
+    packets : list[Packet]
+        packet list
+    met_name : str
+        metadata name to use as epoch time
+
+    Returns
+    -------
+    xr.dataset
+        dataset with all metadata field data in xr.DataArray
+    """
+    metadata_arrays = collections.defaultdict(list)
+    description_dict = {}
+
+    for data_packet in packets:
+        # Add metadata to array
+        for key, value in (data_packet.header | data_packet.data).items():
```

Comment:
Is this going to cause the CCSDS header to be put into a data array?

Reply:
Yes, I have included header in the CDF so far.

Reply:
Is that something SWAPI asked for or is that a standard thing we should do? I wasn't planning to put the headers into the CDFs

Reply:
They didn't ask for this. It's something I have to been carrying information over. But I don't think they will complain if I remove it. If you are thinking of using it and if it helps to remove header, then I am happy to.